src/types/mod.rs

```diff
@@ -716,6 +719,41 @@ impl JsFunction {
             }
         })
     }
+
+    #[cfg(feature = "napi-5")]
+    pub fn new<'a, C, F, V>(cx: &mut C, mut f: F) -> JsResult<'a, JsFunction>
+    where
+        C: Context<'a>,
+        F: FnMut(FunctionContext) -> JsResult<V> + Send + 'static,
+        V: Value,
+    {
+        use std::any;
+        use std::mem;
+        use std::panic::AssertUnwindSafe;
+        use std::ptr;
+
+        use crate::types::error::convert_panics;
+
+        let name = any::type_name::<F>();
```

Comment:
This seemed like a nice little addition without a breaking change. It sets the `Function.prototype.name` to the Rust type name of the function used to create it.

```
[Function: napi_tests::js::strings::return_js_string]
```

It's not the best on closures, but it still seems like an improvement over no name.

```
[Function: napi_tests::main::{{closure}}]
```